specs/UserActivityHistory/UserActivityHistoryAPI.md

```diff
@@ -0,0 +1,243 @@
+UserActivityHistory
+===
+
+# Background
+
+The UserActivity class can be used to note down and preserve a record of activities that the user
+is currently doing on their computer - e.g., browsing a website, reading a Word document, etc.
+
+To record user activity, you use UserActivityChannel to retrieve a UserActivity object via the
+API GetOrCreateUserActivityAsync. If a UserActivity with the given ID already exists, it will be
+returned; otherwise, a new UserActivity object will be created and returned. You can then call
+the API GetSession to return a UserActivitySession object that tracks how long the user is engaged
+in that activity. This structure allows multiple sessions to be associated with the same activity,
+representing the case where the user completes that activity a bit at a time - e.g., beginning to
+watch a movie, then pausing, then watching more later. These will be treated as the same singular
+user activity that spans multiple sessions.
+
+UserActivityHistory is a new set of APIs that allow you to query the past 28 days of the user's
```

Comment:
- can any app query any other app's UserActivity?
- can any apps call this api or is there some sort of capability?

Reply:
Oh, good point, I forgot to include a discussion on this.  An app can query any user activity that has been recorded on the same computer for the same user, but only if a) the user grants permission to access user activity history via a consent dialog, and b) the app is Microsoft logo certified, which basically means that we know what the app is and have given it the OK.  A random exe downloaded off the internet will not be able to access this data.